federation/normalize.go

```diff
@@ -150,8 +150,14 @@ func mergeSameAlias(selections []*graphql.Selection) ([]*graphql.Selection, erro
 		if last == nil || selection.Alias != last.Alias {
 			// Make a copy of the selection so we can modify it below
 			// or when we flatten recursively later.
-			copy := *selection
-			selection = &copy
+			cp := *selection
+			if selection.SelectionSet != nil {
+				// Make a new SelectionSet for the copy so we will not append to the original slice later.
+				cp.SelectionSet = &graphql.SelectionSet{}
+				cp.SelectionSet.Selections = append(cp.SelectionSet.Selections, selection.SelectionSet.Selections...)
+				cp.SelectionSet.Fragments = append(cp.SelectionSet.Fragments, selection.SelectionSet.Fragments...)
+			}
+			selection = &cp
 			newSelections = append(newSelections, selection)
 			last = selection
 			continue
```

Comment:
Optional, followup for below: it looks like we could change `map[string]bool` into `map[string]struct{}`. That might reduce memory usage slightly. Probably a small thing, but maybe useful? Sam for the Fragment map.

Also, the initial capacity of the maps looks suspicious. I wonder if it could be changed and might reduce allocations?
1. They use len(last) but add items proportional to len(selection). Would len(selection) be more appropriate?
2. The Fragment map uses len(...Selections) rather than len(..Fragments). WOuld len(...Fragments) be more appropriate?